code/modules/paperwork/faxmachine.dm

```diff
@@ -0,0 +1,321 @@
+				var/list/combineddepartments = GLOB.alldepartments.Copy()
+				if(long_range_enabled)
+					combineddepartments += GLOB.admin_departments.Copy()
+
+				if(emagged)
+					combineddepartments += GLOB.hidden_admin_departments.Copy()
+
+				destination = input(usr, "To which department?", "Choose a department", "") as null|anything in combineddepartments
+				if(!destination)
+					destination = lastdestination
+		if("auth")
+			if(!is_authenticated && scan)
+				if(check_access(scan))
+					authenticated = TRUE
+			else if(is_authenticated)
+				authenticated = FALSE
+		if("rename")
+			if(copy || photocopy)
+				var/n_name = sanitize(copytext(input(usr, "What would you like to label the fax?", "Fax Labelling", copy.name)  as text, 1, MAX_MESSAGE_LEN))
+				if(usr.stat == 0)
+					if(copy && copy.loc == src)
+						copy.name = "[(n_name ? text("[n_name]") : initial(copy.name))]"
+						copy.desc = "This is a paper titled '" + copy.name + "'."
+					else if(photocopy && photocopy.loc == src)
+						photocopy.name = "[(n_name ? text("[n_name]") : "photo")]"
+
+/obj/machinery/photocopier/faxmachine/proc/handle_cooldown(action, params)
+	sendcooldown = 0
+
+/obj/machinery/photocopier/faxmachine/proc/is_authenticated(mob/user)
+	if(authenticated)
+		return TRUE
+	else if(IsAdminGhost(user))
+		return TRUE
+	return FALSE
+
+/obj/machinery/photocopier/faxmachine/proc/scan(var/obj/item/card/id/card = null)
+	if(scan) // Card is in machine
+		scan.forceMove(get_turf(src))
+		if(!usr.get_active_held_item() && Adjacent(usr))
+			usr.put_in_hands(scan)
+		scan = null
+	else if(Adjacent(usr))
+		if(!card)
+			var/obj/item/I = usr.get_active_held_item()
+			if(istype(I, /obj/item/card/id))
+				if(!usr.drop_item())
+					return
+				I.forceMove(src)
+				scan = I
+		else if(istype(card))
+			if(!usr.drop_item())
+				return
+			card.forceMove(src)
+			scan = card
+
+/obj/machinery/photocopier/faxmachine/verb/eject_id()
+	set category = null
+	set name = "Eject ID Card"
+	set src in oview(1)
+
+	if(usr.restrained())
+		return
+
+	if(scan)
+		to_chat(usr, "You remove [scan] from [src].")
+		scan.forceMove(get_turf(src))
+		if(!usr.get_active_held_item() && Adjacent(usr))
+			usr.put_in_hands(scan)
+		scan = null
+	else
+		to_chat(usr, "There is nothing to remove from [src].")
+
+/obj/machinery/photocopier/faxmachine/proc/sendfax(var/destination,var/mob/sender)
+	if(stat & (BROKEN|NOPOWER))
+		return
+
+	use_power(200)
+
+	var/success = FALSE
+	for(var/thing in GLOB.allfaxes)
+		var/obj/machinery/photocopier/faxmachine/F = thing
+		if( F.department == destination )
+			success = F.receivefax(copy)
+	if(success != FALSE && department != destination)
+		var/datum/fax/F = new /datum/fax()
+		F.name = copy.name
+		F.from_department = department
+		F.to_department = destination
+		F.origin = src
+		F.message = copy
+		F.sent_by = sender
+		F.sent_at = world.time
+		visible_message("[src] beeps, \"Message transmitted successfully.\"")
```

Comment:
All chat sent to Goonchat needs to be marked up with span tags, this'll be really relevant in the future if we try to "dark mode" this window too.